ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/KsqlServerMainTest.java

```diff
@@ -74,4 +85,55 @@ public void shouldStopAppOnErrorStarting() throws Exception {
     // Then:
     verify(executable);
   }
+
+  @Test
+  public void shouldFailIfStreamsStateDirectoryDoesNotExist() {
+    // Given:
+    when(mockStreamsStateDir.exists()).thenReturn(false);
+    when(mockStreamsStateDir.getPath()).thenReturn("/var/lib/kafka-streams");
+
+    expectedException.expect(KsqlServerException.class);
+    expectedException.expectMessage(
+        "The kafka streams state directory does not exist: /var/lib/kafka-streams\n"
+            + " Make sure the directory exists and is writable for KSQL server.");
+
+    // When:
+    KsqlServerMain.enforceStreamStateDirAvailability(mockStreamsStateDir);
+
+  }
+
+  @Test
+  public void shouldFailIfStreamsStateDirectoryIsNotDirectory() {
+    // Given:
+    when(mockStreamsStateDir.exists()).thenReturn(true);
+    when(mockStreamsStateDir.exists()).thenReturn(false);
+    when(mockStreamsStateDir.getPath()).thenReturn("/var/lib/kafka-streams");
+
+    expectedException.expect(KsqlServerException.class);
+    expectedException.expectMessage(
+        "The kafka streams state directory does not exist: /var/lib/kafka-streams\n"
+            + " Make sure the directory exists and is writable for KSQL server.");
+
+    // When:
+    KsqlServerMain.enforceStreamStateDirAvailability(mockStreamsStateDir);
+  }
+
+  @Test
+  public void shouldFailIfStreamsStateDirectoryIsNotWritable() {
+    // Given:
+    when(mockStreamsStateDir.exists()).thenReturn(true);
```

Comment:
nit: we can clean up these tests by having
```
when(mockStreamsStateDir.exists()).thenReturn(true);
when(mockStreamsStateDir.isDirectory()).thenReturn(true);
when(mockStreamsStateDir.canWrite()).thenReturn(true);
when(mockStreamsStateDir.getPath()).thenReturn("/var/lib/kafka-streams");
```
in the setup method so that each individual test only needs to update the single statement relevant to the particular test. Hooray for Mockito! :)

Reply:
Moved them to the setup as you suggested.